extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java

```diff
@@ -186,12 +186,24 @@ DataSegment uploadDataSegment(
 
     final DataSegment outSegment = segment
         .withSize(size)
-        .withLoadSpec(this.makeLoadSpec(new URI(azurePath)))
+        .withLoadSpec(this.makeLoadSpec(segmentConfig.getContainer(), azurePath))
         .withBinaryVersion(binaryVersion);
 
     log.debug("Deleting file [%s]", compressedSegmentData);
     compressedSegmentData.delete();
 
     return outSegment;
   }
+
+  private Map<String, Object> makeLoadSpec(String container, String prefix)
```

Comment:
Could you make the other `makeLoadSpec` delegate to this method instead of duplicating it?

```
  @Override
  public Map<String, Object> makeLoadSpec(URI uri)
  {
    return makeLoadSpec(segmentConfig.getContainer(), uri.toString());
  }
```

Also, is it necessary to pass `container` in since both invocations are using the same value (if you make the suggested modification)?

Reply:
done